src/v3/src/hooks/useInteractionCodeFlow.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { AuthSdkError, IdxStatus, IdxTransaction } from '@okta/okta-auth-js';
+import { useEffect, useState } from 'preact/hooks';
+
+import { RenderResult } from '../../../types';
+import { getTypedOAuthError, RecoverableError } from '../../../util/OAuthErrors';
+import { redirectTransformer } from '../transformer/redirect';
+import { createForm } from '../transformer/utils';
+import { FormBag, InfoboxElement, WidgetProps } from '../types';
+import {
+  getMode, isAuthClientSet, isOauth2Enabled, loc,
+} from '../util';
+
+export const useInteractionCodeFlow = (
```

Comment:
On why we had to make these changes, see the latest in [interactionCodeFlow.js](https://github.com/okta/okta-signin-widget/blob/master/src/v2/client/interactionCodeFlow.js)